File: benchmarks/runtime/branchy_control_flow.lua
local n = 1500000
local i = 0
local sum = 0
while i < n do
  local k = i & 3
  if k == 0 then
    sum = sum + 3
  elseif k == 1 then
    sum = sum + 7
  elseif k == 2 then
    sum = sum + 11
  else
    sum = sum + 13
  end
  if (i & 15) == 0 then
    sum = sum - 1
  elseif (i & 7) == 0 then
    sum = sum + 2
  end
  i = i + 1
end
if sum <= 0 then os.exit(1) end
print(sum)

File: benchmarks/runtime/branchy_control_flow.js
let n = 1500000;
let i = 0;
let sum = 0;
while (i < n) {
  switch (i & 3) {
    case 0: sum += 3; break;
    case 1: sum += 7; break;
    case 2: sum += 11; break;
    default: sum += 13; break;
  }
  if ((i & 15) === 0) {
    sum -= 1;
  } else if ((i & 7) === 0) {
    sum += 2;
  }
  i += 1;
}
if (sum <= 0) process.exit(1);
console.log(sum);
